samples/ClassroomLabs/Modules/Library/Tools/LabCreationLibrary.psm1

```diff
@@ -338,29 +338,55 @@ function Publish-Labs {
                 $ConfigObject
             )
 
-            $lacs = $ConfigObject | Select-Object -Property ResourceGroupName, LabAccountName -Unique
+            $lacs = $ConfigObject | Select-Object -Property ResourceGroupName, LabAccountName, SharedGalleryResourceGroupName, SharedGalleryName -Unique
+
             Write-Host "Operating on the following Lab Accounts:"
             Write-Host $lacs
 
             $block = {
-                param($path, $ResourceGroupName, $LabAccountName)
+                param($path, $ResourceGroupName, $LabAccountName, $SharedGalleryResourceGroupName, $SharedGalleryName)
 
                 Set-StrictMode -Version Latest
                 $ErrorActionPreference = 'Stop'
 
                 $modulePath = Join-Path $path '..\Az.LabServices.psm1'
                 Import-Module $modulePath
 
-                if ((Get-AzLabAccount -ResourceGroupName $ResourceGroupName -LabAccountName $LabAccountName) -eq $null ){
-                    New-AzLabAccount -ResourceGroupName $ResourceGroupName -LabAccountName $LabAccountName | Out-Null
+                $labAccount = Get-AzLabAccount -ResourceGroupName $ResourceGroupName -LabAccountName $LabAccountName
+
+                if ($labAccount -eq $null ){
+                    $labAccount = New-AzLabAccount -ResourceGroupName $ResourceGroupName -LabAccountName $LabAccountName | Out-Null
+                    Write-Host "$LabAccountName lab account created."
+                }
+                else {
+                    Write-Host "$LabAccountName lab account found - skipping create."
+                }
+
+                if ($SharedGalleryResourceGroupName -ne $null && $SharedGalleryName -ne $null){
+
+                    $gallery = $labAccount | Get-AzLabAccountSharedGallery
+                    if ($gallery -ne $null) {
+                        Write-Host "$LabAccountName lab account already has attached gallery $gallery."
```

Comment:
If it's the wrong gallery that's attached to the lab account, should we change it to what's in the CSV file?  I think currently we don't 'fix' settings in other places - but I'm wondering what the user's expectation would be.  Perhaps we should consider this as a "Upsert" (update/insert) operation when running the scripts for any changes?  The downside is that changing the SIG if there are labs & student VMs means that users couldn't reset the student VMs from SIG anymore...

Reply:
Yes, I was hesitant changing it if it's already set for the reasons you mentioned.  I was thinking that if there are failures of some sort (for example, 503 which i've hit occassionally), which we could also include when there is an attached SIG already, that we could output this in a .csv...similar to what you're planning. When you're done with your code, we can likely reuse it here too.